crates/ra_assists/src/introduce_variable.rs

```diff
@@ -44,7 +44,22 @@ pub(crate) fn introduce_variable(ctx: AssistCtx<impl HirDatabase>) -> Option<Ass
             edit.replace(expr.syntax().range(), buf);
         } else {
             buf.push_str(";");
-            indent.text().push_to(&mut buf);
+
+            // We want to maintain the indent level,
+            // but we do not want to duplicate possible
+            // extra newlines in the indent block
+            for chunk in indent.text().chunks() {
+                if chunk.starts_with("\r\n") {
+                    buf.push_str("\r\n");
+                    buf.push_str(chunk.trim_start_matches("\r\n"));
+                } else if chunk.starts_with("\n") {
+                    buf.push_str("\n");
+                    buf.push_str(chunk.trim_start_matches("\n"));
+                } else {
```

Comment:
I've seen this in a few spots where we look for `\r\n` or `\n`. Is there a crate (something in std?) that can help with terminators?

Reply:
I tried to look for something in the std or any crates that would help dealing with terminators but could not find any.

Reply:
I think we should ideally convert \r\n to \n somewhere on the VFS layer, or maybe when we are constructing a syntax tree. Internals should always use \n as a line separator.

Will make an issue for that.

Reply:
https://github.com/rust-analyzer/rust-analyzer/issues/775